src/consensus/merkle.cpp

```diff
@@ -43,11 +43,11 @@
 
 
 uint256 ComputeMerkleRoot(std::vector<uint256> hashes, bool* mutated) {
-    bool mutation = false;
+    if (mutated) *mutated = false;
     while (hashes.size() > 1) {
-        if (mutated) {
```

Comment:
Just `&& !mutation` here?

Reply:
There are two optimizations included here:
1. stop wasting computation time once mutated gets set
2. use the parameter value directly instead of local copy

While the second saves one compare+assign operation and might be insignificant, the first effectively saves useless processing as the result is already known.

To me it is not only optimization but causing a trigger that something was forgotten when reading the code.

@practicalswift : would you support @promag's suggestion to limit the proposal to include only the first part, which would limit the change to one line and gain the most potential savings, or do you oppose the change generally?